### PR TITLE
Register

### PR DESCRIPTION
**Pull request by JohnnatanEstevez**
Add register

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,8 +1,18 @@
-
+import UserRegister from "./componentes/register";
+import UserLogin from "./componentes/login";
+import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
 function App() {
   return (
     <div className="App">
-      <h1>My final project</h1>
+      {/*       <h1>My final project</h1>
+       */}{" "}
```
Comment (davidsa) [inline src/App.js]: remove unused comment

File: src/App.js [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+      <Router>
+        <Routes>
+          <Route path="/" element={<UserRegister />} />
```
Comment (davidsa) [inline src/App.js]: Maybe rename `UserRegister` into `Register` instead. Also for UserLogin

File: src/componentes/register.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+import Button from "react-bootstrap/Button";
+import { Form } from "react-bootstrap";
+import { useDispatch } from "react-redux";
+import { Formik } from "formik";
+import * as yup from "yup";
+import { Link } from "react-router-dom";
+import registerImg from "../images/register.png";
+
+export default function UserRegister() {
+  const schema = yup.object().shape({
```
Comment (davidsa) [inline src/componentes/register.js]: move the schema outside the component. You could also create a directory with all schemas.

File: src/componentes/register.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  });
+
+  const dispatch = useDispatch();
+
+  return (
+    <Formik
+      validationSchema={schema}
+      onSubmit={({ confirmPassword, ...values }, { resetForm }) => {
+        console.log(values);
+        console.log("1.dispatch thunk in register component");
```
Comment (davidsa) [inline src/componentes/register.js]: remove console.logs

File: src/componentes/register.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+        dispatch(thunkUserRegister(values));
+        //resetForm();
```
Comment (davidsa) [inline src/componentes/register.js]: remove comment

File: src/componentes/register.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        isSubmitting,
+        values,
+        touched,
+        isValid,
+        errors,
+      }) => (
+        <div className="container">
+          <Form noValidate onSubmit={handleSubmit}>
+            <h1 className="text-center">Register User </h1>
+            {/*             <img className="img-fluid w-100" src={registerImg} alt="register" />
```
Comment (davidsa) [inline src/componentes/register.js]: remove comment

File: src/componentes/register.js [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+                {errors.confirmPassword}
+              </Form.Control.Feedback>
+            </Form.Group>
+
+            <div className="d-grid gap-2">
+              <Button
+                variant="primary"
+                type="submit"
+                size="lg"
+                //disabled={isSubmitting}
```
Comment (davidsa) [inline src/componentes/register.js]: this comment should be uncommented, is very useful to have the button disabled when submitting

File: src/redux/index.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { createStore, combineReducers, applyMiddleware } from "redux";
+import thunk from "redux-thunk";
+import { registerReducer } from "./users/reducer";
+const reducer = combineReducers({
+  users: registerReducer,
```
Comment (davidsa) [inline src/redux/index.js]: change `users` into `user`

File: src/redux/index.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+});
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
+
+export default createStore(
+  reducer,
+  composeEnhancers(applyMiddleware(thunk))
+  //window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
```
Comment (davidsa) [inline src/redux/index.js]: remove comment

File: src/redux/users/reducer.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import * as TYPES from "./types";
+import { toast } from "react-toastify";
+
+const initialSatate = {
+  user: [],
+  hasError: false,
+  errorMessage: "",
+};
+//const initialSatate = [];
```
Comment (davidsa) [inline src/redux/users/reducer.js]: remove comment

File: src/redux/users/reducer.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+export function registerReducer(state = initialSatate, action) {
+  switch (action.type) {
+    /* case TYPES.REGISTER_USER:
+      //console.log("action register user en reducer user");
+      return [...state, action.payload]; */
```
Comment (davidsa) [inline src/redux/users/reducer.js]: remove comment

File: src/redux/users/reducer.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+    case TYPES.USER_REGISTER_SUCCESS:
+      console.log("4.reducer SUCCESS");
```
Comment (davidsa) [inline src/redux/users/reducer.js]: remove console.log

File: src/redux/users/thunk.js [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+      .then((res) => {
+        console.log(res);
+        console.log("3.thunk SUCCESS");
+        dispatch(userRegisterSuccess(res));
+      })
+      .catch((error) => {
+        console.log("3.1 thunk ERROR");
+        dispatch(userRegisterError(error));
+      });
+  };
```
Comment (davidsa) [inline src/redux/users/thunk.js]: remove console.logs

File: src/redux/users/types.js [hunk 14]
```diff
@@ -0,0 +1,8 @@
+export const USER_REGISTER_STARTED = "USER_REGISTER_STARTED";
+export const USER_REGISTER_SUCCESS = "USER_REGISTER_SUCCESS";
+export const USER_REGISTER_ERROR = "USER_REGISTER_ERROR";
+
+/* export const LOGIN_USER = "LOGIN_USER_STRATED";
+export const LOGIN_USER = "LOGIN_USER_SUCCESS";
+export const LOGIN_USER = "LOGIN_USER_ERROR";
+ */
```
Comment (davidsa) [inline src/redux/users/types.js]: remove commented lines

**Review by davidsa — CHANGES_REQUESTED** [review]
Minor comments, I love the work!!

File: src/App.js (hunk 1 above)
Comment (JohnnatanEstevez) [reply src/App.js]: ok. done!

File: src/App.js (hunk 2 above)
Comment (JohnnatanEstevez) [reply src/App.js]: ok. done!

File: src/componentes/register.js (hunk 4 above)
Comment (JohnnatanEstevez) [reply src/componentes/register.js]: ok. done!

File: src/componentes/register.js (hunk 5 above)
Comment (JohnnatanEstevez) [reply src/componentes/register.js]: ok. done!

File: src/componentes/register.js (hunk 6 above)
Comment (JohnnatanEstevez) [reply src/componentes/register.js]: ok. done!

File: src/componentes/register.js (hunk 7 above)
Comment (JohnnatanEstevez) [reply src/componentes/register.js]: ok. done!

File: src/redux/index.js (hunk 8 above)
Comment (JohnnatanEstevez) [reply src/redux/index.js]: ok. done!

File: src/redux/index.js (hunk 9 above)
Comment (JohnnatanEstevez) [reply src/redux/index.js]: ok. done!

File: src/redux/users/reducer.js (hunk 10 above)
Comment (JohnnatanEstevez) [reply src/redux/users/reducer.js]: ok. done!

File: src/redux/users/reducer.js (hunk 11 above)
Comment (JohnnatanEstevez) [reply src/redux/users/reducer.js]: ok. done!

File: src/redux/users/reducer.js (hunk 12 above)
Comment (JohnnatanEstevez) [reply src/redux/users/reducer.js]: ok. done!

File: src/redux/users/thunk.js (hunk 13 above)
Comment (JohnnatanEstevez) [reply src/redux/users/thunk.js]: ok. done!

File: src/redux/users/types.js (hunk 14 above)
Comment (JohnnatanEstevez) [reply src/redux/users/types.js]: ok. done!

File: src/componentes/register.js (hunk 3 above)
Comment (JohnnatanEstevez) [reply src/componentes/register.js]: Done! schema outside the component

File: src/schemas/schema.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+  password: yup
+    .string()
+    .required("Password is required")
+    .matches(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{6,})/,
+      "Must Contain min 6 Characters, One Uppercase, One Lowercase and One Number"
+    ),
+};
+
+export const schemaRegister = yup.object().shape({
```
Comment (davidsa) [inline src/schemas/schema.js]: registerSchema

File: src/schemas/schema.js [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+    .required()
+    .matches(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{6,})/,
+      "Must Contain min 6 Characters, One Uppercase, One Lowercase and One Number"
+    )
+    .oneOf([yup.ref("password"), null], "Passwords must match")
+    .required("Confirm password is required"),
+});
+
+export const schemaLogin = yup.object().shape({
```
Comment (davidsa) [inline src/schemas/schema.js]: loginSchema

File: src/schemas/schema.js [hunk 17]
```diff
@@ -0,0 +1,30 @@
+import * as yup from "yup";
+export const schema = {
```
Comment (davidsa) [inline src/schemas/schema.js]: basicAuthSchema

**Review by davidsa — COMMENTED** [review]
the naming of some variables could be a little bit better

File: src/schemas/schema.js (hunk 15 above)
Comment (JohnnatanEstevez) [reply src/schemas/schema.js]: Thanks, change done!

File: src/schemas/schema.js (hunk 17 above)
Comment (JohnnatanEstevez) [reply src/schemas/schema.js]: Thanks, change done!

File: src/schemas/schema.js (hunk 16 above)
Comment (JohnnatanEstevez) [reply src/schemas/schema.js]: Thanks, change done!

File: src/componentes/register.js [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+import { registerSchema } from "../schemas/schema";
+
+export default function Register() {
+  const dispatch = useDispatch();
+
+  return (
+    <Formik
+      validationSchema={registerSchema}
+      onSubmit={({ confirmPassword, ...values }, { resetForm }) => {
+        console.log(values);
```
Comment (davidsa) [inline src/componentes/register.js]: remove console.log

**Review by davidsa — APPROVED** [review]

**Pull request merged**